proposal-EDGI_ML_project.md

```diff
@@ -0,0 +1,156 @@
+3. Identify missing data and information files and attempt to find archived versions on other websites.
+
+**Describe how your project meets this need** :
+
+The changes worth reviewing will tend to have some characteristic features which can be classified with the help of various classification algorithms. A considerable amount of time will be devoted to the preparation of the training dataset which is an essential task while working on Machine Learning problems.
+
+A variety of Machine Learning models will be trained on the thoroughly labeled dataset. They will also be tested to evaluate the performance by using some part of the data for validation.
+
+The project will follow the following order of tasks to fill the needs:
+
+1. Step zero will be converting data from different sources (PageFreezer, Versionista) into a common form and storing them in one place.
+2. The first step will be improving the method of computing differences. This will consist of two sub-tasks:
+  1. The changes can be categorized based on the type of object that has changed (text, image, hyperlinks etc.). The first sub-task will be to separate the changes into categories.
+  2. The second sub-task will be to extract meaningful features from the data of each type. This will include extracting information like type and length of the changed matter in a given text. If required, complex techniques can be used to extract features (NLP, Image Processing).
+3. The second task will be writing scripts for different Machine Learning Classification Algorithms which will learn to prioritize the changes. Each script will consist of four basic parts:
+  1. Data loading and pre-processing (if required) to convert data to input format of algorithm. Also, division of data into training and validation sets.
+  2. Machine Learning model creation using libraries like numpy, scikit-learn, [Tensorflow](https://www.tensorflow.org/) etc..
+  3. Evaluation of model performance (using evaluation metrics) and creation of visualizations from the results of the validation set.
+  4. Storing the trained model by serializing the model with the help of modules like pickle, joblib.
+4. The final task will include the following:
+  1. Retrieving the trained models by de-serializing the stored files and using them to prioritize new changes.
+  2. Providing a list of prioritized changes to analysts for reviewing and adding annotations. An important step here will be to add correctly predicted cases to the training dataset. The model will be re-trained only after a sufficient correctly predicted data is added.
+
+**Milestones/Timeline** :
+
+My summer vacation is from May 5 to July 23. I plan to complete the major tasks by the end of July. I will finish the remaining tasks in August and try to solve issues raised (if any) by other members. I consider the completion of each of the main tasks mentioned above as a milestone.
+
+I plan to follow the following tentative timeline:
```

Comment:
Thank you for the time break down with the timeline! This is really helpful to understand how much time you see stuff taking!